inform7/Internal/Inter/BasicInformKit/Sections/Startup.i6t

```diff
@@ -0,0 +1,115 @@
+Startup Template.
+
+The startup rules prepare everything that needs to happen for the Inform runtime.
+
+@h Summary.
+These rules perform the essential tasks required for setting up the virtual
+machine. This includes checking that we are running in a compatible interpreter,
+and setting up various built in Glk objects (when running in a Glk interpreter).
+
+These rules here are the ones which get the basic machinery working
+to the point where it is safe to run arbitrary I7 source text. They necessarily
+do very low-level things, and it is not guaranteed that I7 phrases will behave
+to specification if executed before these early rules have finished. So it
+is hazardous to obstruct or alter them.
+
+Arrangements are a little different here on the Z-machine, because some
+data is retained in the case of a restart.
+
+(Many thanks are due to Eliuk Blau, who found several tricky timing errors
+here and elsewhere in the Glulx-specific code. Frankly, I feel like hanging
+a sign on the following routines which reads "Congratulations on bringing
+light to the Dark Room.")
+
+@h The startup sequence.
+As of IE-0022 we have a simplified startup sequence. |Main| calls this |Startup|
+function, which does two things: run an Architecture16Kit/Architecture32Kit function
+|VM_Check_Functionality| which checks the VM meets the minimum requirements for Inform
+(in Glulx it would test a few Glulx/Glk gestalts; in Z-Code it would probably do
+nothing.) Then it runs the Startup Rules. And that's it - everything else in the
+startup sequence is now in Inform-accessible rules.
+
+=
+[ Startup;
+	VM_Check_Functionality();
+	FollowRulebook(STARTUP_RB);
+	rfalse;
+];
+
+@h Initialise Memory Rule.
+This rule amalgamates some minimal initialisations which all need to happen
+before we can risk using some of the more exotic I7 kinds:
+
+(a) The language definition might call for initialisation, although the
+default language of play (English) does not.
+
+(b) We start the machinery needed to check that property accesses are
+valid during play.
+
+(c) We initialise the memory allocation heap, and expand the literal
+constants, as hinted above: these are called "block constants" since
+they occupy blocks of memory.
+
+(d) Some platform-specific dictionary and parse buffer related globals
+are initialised.
+
+=
+[ INITIALISE_MEMORY_R;
+	LanguageInitialise();
+
+	HeapInitialise(); ! Create a completely unused memory allocation heap
+	StackFramingInitialise(); ! Create an empty stack
+	CreateDynamicRelations(); ! Create relation structures on the heap
+
+	#Ifdef TARGET_GLULX;
```

Comment:
`INITIALISE_MEMORY_R` and `SEED_RANDOM_NUMBER_GENERATOR_R` contain two conditional compilation blocks with code that were originally in the two architectures' `VM_Initialise` functions. It seemed neater to me to group all this code together, rather than putting it in functions in the architecture kits. But we could do that if you'd prefer.